main.go

```diff
@@ -199,6 +199,10 @@ func main() { //nolint:funlen
 		setupLog.Error(err, "unable to update from legacy operator version")
 	}
 
+	if err = upgrade.CleanupExistingResource(mgr.GetConfig(), dscApplicationsNamespace); err != nil {
```

Comment:
So this will be invoked every time the pod is created... is that what we want?

Reply:
i think this probably is the least invoking by restart operator pod then in dashboard reconcile loop.

but we do need to work on a more generic way to deal with "prune"
also thinking, this logic can be removed in 2.7
as the normal upgrade is 2.4(with CRD)->2.5(with CRD)->2.6(remove it)->2.7(no need logic)

Reply:
Ok, so this is left-over from operator v1 world then?
```
Group:    "dashboard.opendatahub.io",
Version:  "v1",
Resource: "odhapplications",
```

Reply:
not v1, even in the v2.0->v2.5, but the plan is dashboard want to remove this CRD from their manifests in 2.6.